core/src/main/java/io/questdb/griffin/engine/functions/table/ReadParquetRecordCursor.java

```diff
@@ -394,6 +396,27 @@ public int getVarcharSize(int col) {
             return VarcharTypeDriver.getValueSize(auxPtr, currentRowInRowGroup);
         }
 
+        private DirectBinarySequence bsView(int columnIndex) {
+            if (bsViews.getQuiet(columnIndex) == null) {
+                bsViews.extendAndSet(columnIndex, new DirectBinarySequence());
```

Comment:
This is not thread-safe. Is there a guarantee that the record will not be used to get different column values from multiple threads, like parallel group by, filtering?

Reply:
Yes, most, if not all of our records are thread unsafe as they're always accessed by a single thread. In case of parallel group by/filter worker threads use their own (thread-local) records.